### PR TITLE
Fixes in RasterYuv420Planar class

### PR DESCRIPTION
**Pull request by sebastianbaginski**
Adds test case for `RasterYuv420Planar`
Corrected buffer ownership in fallback `open` method in `RasterImage`
Added missing `free` in `RasterYuvPlanar`
@marcusnaslund

### CODE REVIEW

**Comment by marcusnaslund** [comment]
Test fails in magic.

**Comment by sebastianbaginski** [comment]
magic :+1: 
> sbaginski@mosquito:~/versioned2/ooc/ooc-kean$ magic
> -> magic 0.1.20-alpha
> sbaginski@mosquito:~/versioned2/ooc/ooc-kean$ 

I think it simply failed during tests. Updated, well see now

**Comment by marcusnaslund** [comment]
```
11:53:53 RasterYuv420Planar [lang/Backtrace] Falling back on execinfo.. (build extension if you want fancy backtraces)
f. failed in 0.64s, total: 17.09s
  -> 'convert' (expect: 1)
```

**Comment by sebastianbaginski** [comment]
Works locally though, weird.

**Comment by marcusnaslund** [comment]
Circle, like rock, works in mysterious ways.

**Comment by sebastianbaginski** [comment]
Well I think I will maybe just skip the distance test.

**Comment by sebastianbaginski** [comment]
@marcusnaslund : it works now.

**Comment by marcusnaslund** [comment]
What was wrong?

**Comment by sebastianbaginski** [comment]
`distance` between test images was not zero on Circle but zero locally

File: source/draw/RasterImage.ooc [hunk 1]
```diff
@@ -62,14 +62,16 @@ RasterImage: abstract class extends Image {
 		x, y, imageComponents: Int
 		data := StbImage load(filename, x&, y&, imageComponents&, 0)
 		result: This
+		buffer := ByteBuffer new(data as UInt8*, x * y * imageComponents, true)
 		match (imageComponents) {
 			case 1 =>
-				result = RasterMonochrome new(ByteBuffer new(data as UInt8*, x * y * imageComponents), IntVector2D new (x, y))
+				result = RasterMonochrome new(buffer, IntVector2D new (x, y))
```
Comment (marcusnaslund) [inline source/draw/RasterImage.ooc]: While we're at it, there should be no space after `new` here or below.

Comment (sebastianbaginski) [inline source/draw/RasterImage.ooc]: updated

**Pull request merged**